lightning/src/chain/channelmonitor.rs

```diff
@@ -2351,13 +2685,39 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 								// resolve the source HTLC with the original sender.
 								payment_data = Some(((*source).clone(), htlc_output.payment_hash, htlc_output.amount_msat));
 							} else if !$holder_tx {
-									check_htlc_valid_counterparty!(self.current_counterparty_commitment_txid, htlc_output);
+								check_htlc_valid_counterparty!(self.current_counterparty_commitment_txid, htlc_output);
 								if payment_data.is_none() {
 									check_htlc_valid_counterparty!(self.prev_counterparty_commitment_txid, htlc_output);
 								}
 							}
 							if payment_data.is_none() {
 								log_claim!($tx_info, $holder_tx, htlc_output, false);
+								let outbound_htlc = $holder_tx == htlc_output.offered;
+								if !outbound_htlc || revocation_sig_claim {
+									self.onchain_events_awaiting_threshold_conf.push(OnchainEventEntry {
+										txid: tx.txid(), height,
+										event: OnchainEvent::HTLCSpendConfirmation {
+											input_idx: input.previous_output.vout,
+											preimage: if accepted_preimage_claim || offered_preimage_claim {
+												Some(payment_preimage) } else { None },
+											// `payment_data.is_none()` implies that this is a payment
+											// to us, as we haven't learned anything to cause us to
+											// update another channel or our offchain state. Thus, wait
+											// for the CSV delay before dropping the HTLC from
+											// claimable balance if the claim was an HTLC-Success or
+											// HTLC-Timeout transaction.
+											on_to_local_output_csv: if accepted_preimage_claim || offered_timeout_claim {
+												Some(self.on_holder_tx_csv) } else { None },
```

Comment:
Oh wait sorry... So if this is a payment to us, and it's an `offered_timeout_claim` (implying counterparty commit tx), why would we use `on_holder_tx_csv`? wouldn't we want `on_counterparty_tx_csv`?

Reply:
Its helpful that the `offered_timeout_claim` case's `Some` value here is dead code :p. If its an inbound (ie from counterparty to us) HTLC, claimed via a timeout on a counterparty commitment transaction, it is not our HTLC/we won't have a preimage for it and we won't care about it in `get_claimable_balance`. All that said, yea, its wrong.